projects/cdk/schematics/ng-update/steps/rename-types.ts

```diff
@@ -1,38 +1,58 @@
+import {TypeNode} from 'ts-morph';
+import {getImports, ImportSpecifier, Node} from 'ng-morph';
+
 import {getNamedImportReferences} from '../../utils/get-named-import-references';
-import {ImportSpecifier, Node} from 'ng-morph';
 import {TYPES_TO_RENAME} from '../constants/types';
-import {removeImport} from '../../utils/remove-import';
+import {removeImport, renameImport} from '../../utils/import-manipulations';
 
 export function renameTypes(): void {
-    TYPES_TO_RENAME.forEach(({from, to, moduleSpecifier}) => {
-        renameType(from, to, moduleSpecifier);
+    TYPES_TO_RENAME.forEach(({from, to, moduleSpecifier, preserveGenerics}) => {
+        renameType(from, to, moduleSpecifier, preserveGenerics);
     });
 }
 
 function renameType(
     from: string,
     to?: string,
     moduleSpecifier?: string | string[],
+    preserveGenerics: boolean = false,
 ): void {
     const references = getNamedImportReferences(from, moduleSpecifier);
 
     references.forEach(ref => {
         const parent = ref.getParent();
 
-        if (Node.isImportSpecifier(parent) && to) {
-            renameImport(parent, to, from);
-        } else if (Node.isImportSpecifier(parent) && !to) {
-            removeImport(parent);
+        if (Node.isImportSpecifier(parent)) {
+            renameTypeImport(parent, from, to);
```

Comment:
nit:  function not only renames now - `renameOrRemoveimport` or `processImport` (or something else)?

Reply:
fixed